05_classify_mutations_multimodal/nbconverted/plot_multimodal_results.py

```diff
@@ -0,0 +1,304 @@
+u_results_df_4.drop(columns='n_dims', inplace=True)
+
+# make sure data loaded matches our expectations
+print(u_results_df_1.training_data.unique())
+print(u_results_df_2.training_data.unique())
+print(u_results_df_3.training_data.unique())
+print(u_results_df_4.training_data.unique())
+print(u_results_df_1.seed.unique())
+print(u_results_df_2.seed.unique())
+print(u_results_df_3.seed.unique())
+print(u_results_df_4.seed.unique())
+
+
+# In[7]:
+
+
+# first, take the unimodal results and the multimodal results,
+# and concatenate them into one giant results df
+all_results_df = pd.concat((results_df,
+                            u_results_df_1,
+                            u_results_df_2,
+                            u_results_df_3,
+                            u_results_df_4))
+
+print(all_results_df.shape)
+print(all_results_df.training_data.unique())
+all_results_df.head()
+
+
+# In[8]:
+
+
+# then, for each training data type, get the AUPR difference between signal and shuffled
+compare_df = pd.DataFrame()
+for training_data in all_results_df.training_data.unique():
+    data_compare_df = au.compare_control_ind(
+        all_results_df[all_results_df.training_data == training_data],
+        identifier='identifier',
+        metric='aupr',
+        verbose=True
+    )
+    data_compare_df['training_data'] = training_data
+    data_compare_df.rename(columns={'identifier': 'gene'}, inplace=True)
+    compare_df = pd.concat((compare_df, data_compare_df))
+    
+compare_df.head(10)
+
+
+# In[9]:
+
+
+# each subplot will show results for one gene
+sns.set({'figure.figsize': (24, 12)})
+fig, axarr = plt.subplots(2, 4)
+compare_df.sort_values(by=['gene', 'training_data'], inplace=True)
+compare_map = {v: str(i) for i, v in enumerate(compare_df.training_data.unique())}
+compare_df['train_index'] = compare_df.training_data.map(compare_map)
+min_aupr = compare_df.delta_aupr.min()
+max_aupr = compare_df.delta_aupr.max()
+
+
+data_order =['expression',
+             'me_27k',
+             'me_450k',
+             'expression.me_27k',
+             'expression.me_450k',
+             'me_27k.me_450k',
+             'expression.me_27k.me_450k']
+
+# plot mean performance over all genes in pilot experiment
+for ix, gene in enumerate(results_df.identifier.unique()):
+    
+    ax = axarr[ix // 4, ix % 4]
+    
+    plot_df = compare_df[(compare_df.gene == gene)]
+
+    sns.boxplot(data=plot_df, x='training_data', y='delta_aupr',
+                order=data_order, ax=ax)
+    ax.set_title('Prediction for {} mutation'.format(gene))
+    ax.set_xlabel('Training data type')
+    # hide x-axis tick text
+    ax.get_xaxis().set_ticklabels([])
+    ax.set_ylabel('AUPR(signal) - AUPR(shuffled)')
+    ax.set_ylim(-0.2, max_aupr)
+    
+handles = []
+for ix, data in enumerate(data_order):
+    handle = mpatches.Patch(color=sns.color_palette()[ix], label=data)
+    handles.append(handle)
+    
+fig.delaxes(axarr[1, 3])
+plt.legend(handles=handles, loc='lower right')
```

Comment:
It's interesting that methyl27k does so much better than the other data types for TET2, but performs the worst for almost all other genes.

Reply:
Yeah, I agree. I looked at TET2 because it's directly involved in DNA methylation (or more accurately DNA demethylation), but I'm hesitant to draw many conclusions from it because it's not that frequently mutated.

The cancer types that meet our cutoff for TET2 mutation (>10 samples mutated and >5% of samples mutated per cancer type) are  SKCM (31 samples, 8%) and UCEC (47 samples, 9%). SKCM has a very high baseline mutation burden so I'm not super convinced any of those samples are strongly driven by TET2 mutation (could be just chance/passenger mutations), and I can't find much work linking TET gene mutation to uterine/endometrial cancers (maybe [this](https://doi.org/10.1177/1010428317695017)?), although that association seems more plausible to me.

I still think it's interesting that performance is better with the methylation data here (since that's what we'd expect based on the function of TET2), but I'm interpreting with a bit of caution.

Reply:
Interesting! I think cautious interpretations are fine here.